test/builtin/test_patterns.py

```diff
@@ -26,3 +26,17 @@ def test_replace_all():
         ),
     ):
         check_evaluation(str_expr, str_expected, message)
+
+
+def test_rule_repl_cond():
+    for str_expr, str_expected, message in (
+        ("f[x]/.(f[u_]->u^2/; u>3/; u>2)", "f[x]", "conditions are not evaluated"),
```

Comment:
I just tried this on galatea and I don't see these results. Here is what I see:  

```
>> f[x]/.(f[u_]->u^2/; u>3/; u>2)
 = x ^ 2 /; x > 3 /; x > 2
>> f[1.]/.(f[u_]->u^2/; u>3/; u>2)
 = 1. /; 1. > 3 /; 1. > 2
```

Reply:
You are right. When I tested it, I did it with delayed rules instead. There is something that I still am missing.